src/lib/optimizer/strategy/stored_table_column_alignment_rule.cpp

```diff
@@ -0,0 +1,63 @@
+#include "optimizer/strategy/stored_table_column_alignment_rule.hpp"
+
+#include <algorithm>
+#include <memory>
+
+#include "logical_query_plan/abstract_lqp_node.hpp"
+
+namespace opossum {
+
+void StoredTableColumnAlignmentRule::apply_to(const std::shared_ptr<AbstractLQPNode>& root) const {
+  StoredTableNodeUnorderedMap<std::vector<std::shared_ptr<StoredTableNode>>> gathered_stored_table_nodes_for_node;
+  StoredTableNodeUnorderedMap<std::vector<ColumnID>> aligned_pruned_column_ids_for_node;
+  recursively_gather_stored_table_nodes(root, gathered_stored_table_nodes_for_node, aligned_pruned_column_ids_for_node);
```

Comment:
Calling a method recursively is not the preferred method of traversing an LQP, as nodes in diamond structures will be visited more than once. Instead, we prefer `visit_lqp`. An additional advantage is that you can simply capture your map(s) by reference.